R/data.R

```diff
@@ -1,68 +1,294 @@
+# limit to 50 characters the description 
+# for the sdg_description
+#############################################
+
+limited_description<-function(){
+  description<-substring(load_all_indicators()$sdg_description, 0, 30)
+  for(i in 1:length(load_all_indicators()$sdg_description)){
+    if(length(load_all_indicators()$sdg_description[i])<25){
+      description[i]<-paste(description[i], '[...]')
+    }
+  }
+  return(description)
+}
+
+############################################
+# Create sdg list for the control panel
+#############################################
+
+sdg_list <- function(){
+  
+  sdg_list <- setNames(unique(indicators$sdg_code),as.character(paste(unique(indicators$sdg_code), unique(limited_description()), sep=': ')))
+  sdg_list[!is.na(sdg_list)]
+  sdg_list <- sort(sdg_list)
+  sdg_list <- append((sdg_list), 'Select All', after = 0)
+  
+  return(sdg_list)
+}
+
+############################################
+# Create sdg code list for the SDG filter
+#############################################
+
+sdg_code_list <- function(){
+  con <- get_sql_connection()
+  
+  sql<-"SELECT sdg_indicators.code as sdg_code
+           
+        FROM sdg_indicators;"
+  
+  sdg_code_df <- dbGetQuery(con, sql)
+  dbDisconnect(con)
+  sdg_code_list <- unique(sdg_code_df$sdg_code)
+  
+  return(sdg_code_list)
+}
+
+#####################################
+# load data by country code
+#####################################
+
+load_indicators<-function(country_code){
+  
+  con <- get_sql_connection()
+  sql <- "SELECT
+         indicators.group_name,
+         indicators.subgroup_name,
+         indicators.indicator_value,
+         datasets.region,
+         datasets.year,
+         datasets.country_code,
+         datasets.description,
+         datasets.population_definition,
+         datasets.source_url,
+         datasets.comment,
+         countries.name as countries_name,
+         sdg_indicators.code as sdg_code
+         FROM indicators
+         LEFT JOIN datasets on datasets.id = indicators.dataset_id
+         LEFT JOIN countries on countries.ISO_code = datasets.country_code
+         LEFT JOIN sdg_indicators on  sdg_indicators.id = indicators.sdg_indicator_id"
+                        
+  if(! is.null(country_code)) {
+    sql <- paste(sql, " WHERE datasets.country_code = '",country_code, "'", sep = "")
+  }
+  
+  indicators <- dbGetQuery(con,paste(sql,";"))
+  dbDisconnect(con)
+  
+  return(indicators)
+}
+
+
+#####################################
+# Create src for the SDG image
+#####################################
+
+show_image<-function(sdg_number){
+  src_image <- paste("images_app/E-WEB-Goal-",sdg_number,".png", sep="")
+  alt_image <- paste("sdg",sdg_number, sep="")
+  renderImage({
+      return(list(
+        width = 100,
+        height = 100,
+        src = src_image,
+        contentType = "image/png",
+        alt = alt_image
+      ))
+    
+    
+  }, deleteFile = FALSE)
+}
```

Comment:
Again, really nice clear function. I'd suggest perhaps a slightly more specific name (e.g. `show_sdg_image`), just in case you have other types of image you want to show in future.